coordinates.cpp

```diff
@@ -1 +1,47 @@
+#include "coordinates.h"
+#include <cmath> 
+#include <iostream>
+#include <vector> 
+#include <utility> 
 
+using namespace std;
+
+double Haversine(Coordinate a, Coordinate b) 
+    {   double rad = 6371; 
+        // find the distance between lat and long 
+        double latDistance = (b.first - a.first) * M_PI / 180.0; 
+        double longDistance = (b.second - a.second) * M_PI / 180.0;
+
+        // convert the distance to radians 
+        b.first = (b.first) * M_PI / 180.0; 
+        a.first = (a.first) * M_PI / 180.0; 
+
+        // find distance 
+        double powFormula; 
```

Comment:
I wonder if any of the problems you mentioned have something to do with this? It may be converting the right number into the wrong units if that makes any sense.